core/app/middleware_test.go

```diff
@@ -7,13 +7,72 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/flapflapio/simulator/internal/simtest"
 	"github.com/obonobo/mux"
 )
 
-func TestTrim(t *testing.T) {
-	type testcase struct{ name, input, output string }
+func TestCORS(t *testing.T) {
+	tests := []struct {
+		name, route, origin string
+		origins             []string
+	}{
+		{
+			name:    "wildcard",
+			route:   "/",
+			origin:  "http://localhost:3000",
+			origins: []string{"*"},
+		},
+		{
+			name:    "localhost",
+			route:   "/some/path",
+			origin:  "http://localhost:3000",
+			origins: []string{"http://localhost:3000"},
+		},
+		{
+			name:   "web app",
+			route:  "/some/path",
+			origin: "https://machinist.flapflap.io",
+			origins: []string{
+				"http://localhost:3000",
+				"https://localhost:3000",
+				"https://machinist.flapflap.io",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 
+			recorder := httptest.NewRecorder()
+			router := mux.NewRouter()
+			router.HandleFunc(tc.route, func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusOK)
+			})
+
+			req := simtest.MustCreateRequest(t, "GET", tc.route, nil)
+			req.Header.Add("Origin", tc.origin)
+
+			handler := CORS(tc.origins...)(router)
+			handler.ServeHTTP(recorder, req)
+
+			// Assertions
+			assertRequestStatus(t, recorder, http.StatusOK)
+			if len(tc.origins) > 1 {
+				assertResponseHeader(t, recorder, "Access-Control-Allow-Origin", tc.origin)
+				assertResponseHeader(t, recorder, "Vary", "Origin")
+			} else if len(tc.origins) == 1 {
+				assertResponseHeader(t, recorder, "Access-Control-Allow-Origin", tc.origins[0])
+			}
+		})
+	}
+}
+
+func TestTrim(t *testing.T) {
 	somePath := "/some/path/"
```

Comment:
Please, you may explain the somePath?

Reply:
This test is for a function that trims trailing slashes from paths (we are using it to trim slashes from routes that are requested, could also be used for file paths).

So the somePath variable is a path to be tested. We are storing it here in a variable because there are some test cases below that build on this variable by adding more slashes